harness/tests/experiment/pytorch/test_pytorch_trial.py

```diff
@@ -734,16 +836,24 @@ def test_checkpoint_loading(ckpt: str, istrial: bool):
         assert isinstance(trial, torch.nn.Module), type(trial)
 
 
-def amp_metrics_test(trial_class, training_metrics):
+def amp_metrics_test(trial_class, training_metrics, agg_freq=1):
     loss_prev = None
     GROWTH_INTERVAL = trial_class._growth_interval
     MIN_SCALED_LOSS_TO_REDUCE_SCALE = 32760
     growth_countdown = GROWTH_INTERVAL
     # Only attempt assertions up to and including the penultimate batch, because
     #  we may not have the updated scale from the final batch.
-    for idx, (metrics, next_metrics) in enumerate(zip(training_metrics[:-1], training_metrics[1:])):
+    for idx, (metrics, next_metrics) in enumerate(
+        zip(
+            training_metrics[:-1],
+            training_metrics[1:],
+        )
+    ):
+        if (idx + 1) % agg_freq != 0:
+            # Don't test mini-batches
```

Comment:
I do think the terms "batch" and "mini-batch" are ambiguous in our field, but I don't think using "mini" batch in this way is more helpful, since the term mini-batch has often been used to describe what we think of as just a "batch".  See [here](https://machinelearningmastery.com/gentle-introduction-mini-batch-gradient-descent-configure-batch-size/) for such an example.

Better to be precise and literal here:

```python
# Only test batches which land on aggregation_frequency boundaries.
```

Reply:
Thanks for the clarification! This is done.